webviz_subsurface/plugins/_structural_uncertainty/figures/intersection.py

```diff
@@ -0,0 +1,324 @@
+    # Mean trace
+    traces.append(
+        {
+            "name": legendname,
+            "y": stat_surfaces["Mean"][:, 1],
+            "x": stat_surfaces["Mean"][:, 0],
+            "mode": "lines",
+            "hoverinfo": "skip",
+            "fill": "tonexty",
+            "fillcolor": fill_color,
+            "line": {"color": line_color},
+            "legendgroup": name,
+            "showlegend": showlegend,
+        }
+    )
+    # P90 trace
+    traces.append(
+        {
+            "name": legendname,
+            "y": stat_surfaces["P90"][:, 1],
+            "x": stat_surfaces["P90"][:, 0],
+            "mode": "lines",
+            "hoverinfo": "skip",
+            "fill": "tonexty",
+            "fillcolor": fill_color,
+            "line": {"width": 0, "color": line_color},
+            "legendgroup": name,
+            "showlegend": False,
+        }
+    )
+    # Minimum trace
+    traces.append(
+        {
+            "name": legendname,
+            "y": stat_surfaces["Min"][:, 1],
+            "x": stat_surfaces["Min"][:, 0],
+            "mode": "lines",
+            "hoverinfo": "skip",
+            "fill": "tonexty",
+            "fillcolor": fill_color,
+            "line": {"width": 0, "color": line_color},
+            "legendgroup": name,
+            "showlegend": False,
+        }
+    )
+    return traces
+
+
+# pylint: disable=too-many-arguments
+@CACHE.memoize(timeout=CACHE.TIMEOUT)
+def get_plotly_trace_realization_surface(
+    surfaceset: SurfaceSetModel,
+    fence_spec: np.ndarray,
+    name: str,
+    legendname: str,
+    attribute: str,
+    realization: int,
+    showlegend: bool = False,
+    sampling: Optional[str] = "billinear",
+    color: str = "red",
+) -> Dict[str, Any]:
+    """Returns a plotly line trace for a surface"""
+    surface = surfaceset.get_realization_surface(
+        name=name, attribute=attribute, realization=realization
+    )
+    fencexy = get_surface_randomline(surface, fence_spec, sampling)
+    return {
+        "x": fencexy[:, 0],
+        "y": fencexy[:, 1],
+        "name": legendname,
+        "text": f"{legendname} realization: {realization}",
+        "showlegend": showlegend,
+        "hoverinfo": "y+x+text",
+        "mode": "lines",
+        "marker": {"color": hex_to_rgba(hex_string=color, opacity=0.5)},
+    }
+
+
+@CACHE.memoize(timeout=CACHE.TIMEOUT)
+def get_plotly_zonelog_trace(
+    well: xtgeo.Well,
+    zonelog: str,
+) -> List[Dict]:
+    """Zonetops are extracted from a zonelog and plotted as markers"""
+
+    df = well.dataframe
+
+    # Find zone transitions
+    df["transitions"] = df[zonelog].diff()
+    logrecord = well.get_logrecord(zonelog)
+    traces = []
+    for idx in range(0, df.shape[0] - 1):
+        # Use current sample if moving upwards in stratigraphy
+        # Use next sample if moving downwards in stratigraphy
+        if (
+            df.iloc[idx + 1]["transitions"] == -1.0
+            or df.iloc[idx]["transitions"] == 1.0
```

Comment:
Will this work if you e.g. have a fault or erosional surface that causes the stratigraphy to not strictly increase/decrease by one along the well?